### PR TITLE
Add compatibility with symfony 3.* and 4.*

### PR DESCRIPTION
**Pull request by pascal-hofmann**
Would be great to have this merged soon.

### CODE REVIEW

**Comment by fabpot** [comment]
We need to be sure that this works on both 3.x and 4.0, so I would also change the `.travis.yml` file to be sure that we are running tests for at least 3.4 and 4.0. You can have a look at `twigphp/twig` for an example of how to make it work.

**Comment by pascal-hofmann** [comment]
@fabpot: Thanks for your feedback, I'll change `.travis.yml`accordingly. 

I fail to see where multiple symfony versions are being used by travis in `twigphp/twig`, and instead will follow what's described here: https://symfony.com/doc/current/setup/bundles.html#update-the-travis-ci-configuration

Somehow the travis builds for this PR don't work. Travis just says "Abuse detected". I've reached out to the support to get this fixed.

**Comment by fabpot** [comment]
Indeed, Silex is a better example: https://github.com/silexphp/Silex/blob/master/.travis.yml

**Comment by pascal-hofmann** [comment]
Closing and re-opening to trigger travis rebuild.

**Comment by pascal-hofmann** [comment]
I finally managed to get the travis "Abuse detected" problem solved and all travis builds passing.

@fabpot Is the PR ok like this?

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,57 @@
...
-  - 5.4
-  - 5.5
-  - 5.6
-  - hhvm
+cache:
+    directories:
+      - $HOME/.composer/cache/files
+
+before_install:
+    # Circumvent a bug in the travis HHVM image - ships with incompatible PHPUnit.
```
Comment (fabpot) [inline .travis.yml]: I would remove everything related to HHVM as we (Symfony) do not support HHVM anymore anyway.

File: composer.json [hunk 2]
```diff
@@ -11,8 +11,8 @@
     ],
     "require": {
         "php": ">=5.3.0",
-        "symfony/http-foundation": "~2.1",
-        "symfony/http-kernel": "~2.1"
+        "symfony/http-foundation": "~2.1|~3.0|~4.0",
+        "symfony/http-kernel": "~2.1|~3.0|~4.0"
```
Comment (fabpot) [inline composer.json]: What about bumping `~2.1` to `~2.7`? `2.1` is very old now, so I don't think we need to support it.

File: tests/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?php
+if (class_exists('PHPUnit_Framework_TestCase') === false && class_exists('PHPUnit\Framework\TestCase') === true) {
```
Comment (fabpot) [inline tests/bootstrap.php]: Small thing, but can you add a blank line just after `<?php`? And add a new line at the end of the file?

Comment (fabpot) [inline tests/bootstrap.php]: Could be simplified to `if (!class_exists('PHPUnit_Framework_TestCase') && class_exists('PHPUnit\Framework\TestCase')) {`

**Comment by pascal-hofmann** [comment]
@fabpot I changed the PR according to your suggestions.

**Pull request merged**